### PR TITLE
feat: parse real-world dope.bat fixture

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary
- Extends the grammar to handle the batch idioms in secretGeek/dod's `dope.bat` (DOS on Dope) so it parses with zero errors.
- Ships `examples/dope.bat` as a CI fixture and asserts zero `ERROR`/`MISSING` nodes alongside `search.bat`.
- Keeps all pre-existing corpus tests green (188/188) and adds 7 new cases covering the new idioms.

## Grammar changes
- `if`/`exist`/`defined` operands accept concatenated token sequences, so `if x'%2'==x'/?'`, `if exist %2\Content`, and `if '%1'=='site'` now parse as single operands instead of bailing out.
- `_if_word` now beats `command_name` via higher token precedence; continuation segments use `token.immediate` so whitespace still terminates operands and the `/i` option still wins over the broadened word pattern.
- `variable_reference` grows patterns for `%%NAME%%`, `%%%%[0-9a-zA-Z]`, `%%%%%%[0-9a-zA-Z]`, and `%%[0-9]` — the percent-escaping forms that appear inside heredoc-style `echo ...>> file`.
- `cmd` gains an `echo`-specific branch aliased back to `argument_list`: its content allows strings with `=`, stray `(`/`)`, and standalone `!` as literal text, while other commands keep their structured argument parsing so `(echo two)` still resolves as a `parenthesized` statement.

## Test plan
- [x] `npx tree-sitter test` — 188/188 passing
- [x] `tree-sitter parse examples/*.bat --quiet --stat` — 9/9 clean
- [x] `tree-sitter parse examples/dope.bat` — 0 ERROR / 0 MISSING
- [x] `npm run lint` — clean
- [x] CI asserts zero errors on both `examples/search.bat` and `examples/dope.bat`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added a new Windows batch MVC scaffolding utility for generating project structures, controllers, and models.

* **Tests**
  * Expanded test coverage for echo command and if statement parsing with new test cases covering edge cases and complex syntax.

* **Chores**
  * Improved parsing logic for echo commands and if statements with enhanced variable reference handling and tokenization.

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -122,10 +122,14 @@ export default grammar({
       $.redirect_stmt, $.pipe_stmt, $.comment,
     ),
     _if_operand: ($) => choice(
-      $.string, $.variable_reference, $.integer, $.bracketed_value,
-      alias($._if_word, $.argument_value),
+      $.string, $.bracketed_value,
+      prec.right(seq(
+        choice($.variable_reference, alias($._if_word, $.argument_value), $.integer),
+        repeat(choice($.variable_reference, alias($._if_word_rest, $.argument_value))),
+      )),
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in 761ce60. Thanks @gemini-code-assist!

File: grammar.js [hunk 2]
```diff
@@ -122,10 +122,14 @@ export default grammar({
...
     ),
-    _if_word: () => token(prec(-1, /[^=<>\s\[\]"|&()][^=<>\s"|&()]*/)),
+    _if_word: () => token(prec(1, /[^=<>\s\[\]"|&()%][^=<>\s"|&()%]*/)),
+    _if_word_rest: () => token.immediate(/[^=<>\s\[\]"|&()%][^=<>\s"|&()%]*/),
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in 32eb63b. Thanks @coderabbitai!

File: grammar.js [hunk 3]
```diff
@@ -218,7 +226,16 @@ export default grammar({
...
+    cmd: ($) => prec.right(5, choice(
+      seq(optional('@'), alias(kw('echo'), $.command_name), optional(alias($._echo_text, $.argument_list))),
+      seq(optional('@'), choice($.command_name, $.variable_reference), optional($.argument_list)),
+    )),
+    _echo_text: ($) => prec.right(repeat1(choice(
+      $.variable_reference,
+      $.string,
+      alias($._echo_literal, $.argument_value),
+    ))),
+    _echo_literal: () => token(prec(-1, /(?:\^[&|<>^()]|[^\s|&><"\r\n%!()])+|[()!]/)),
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in 64bacf8. Thanks @gemini-code-assist!
